File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var arraySign = function(nums) {
    let negCount = 0;

    for(let num of nums){
        if(num == 0) return 0;
        if(num < 0) negCount++;
    }

    return (negCount%2 == 0) ? 1 : -1;
};

File: 1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {
        int negCount = 0;
        
        for(int num : nums){
            if(num == 0) return 0;
            if(num < 0) negCount++;
        }
        
        if(negCount%2 == 0) return 1;
        else return  -1;
    }
}